cmd/deployment-check/run_check.go

```diff
@@ -0,0 +1,290 @@
+	select {
+	case serviceResult = <-createService(serviceConfig):
+		// Handle errors when the service creation process completes.
+		if serviceResult.Err != nil {
+			log.Errorln("error occurred creating service in cluster:", serviceResult.Err)
+			reportErrorsToKuberhealthy([]string{serviceResult.Err.Error()})
+			return nil
+		}
+		// Continue with the check if there is no error.
+		log.Infoln("Created service in", serviceResult.Service.ObjectMeta.Namespace, "namespace:", serviceResult.Service.ObjectMeta.Name)
+	case <-ctx.Done():
+		// If there is a cancellation interrupt signal, exit.
+		log.Infoln("Cancelling check and shutting down due to interrupt.")
+		return nil
+	case <-runTimeout:
+		// If creating a service took too long, exit.
+		reportErrorsToKuberhealthy([]string{"failed to create deployment within timeout"})
+		return nil
+	}
+
+	// Get an ingress hostname associated with the service.
+	hostname := getServiceLoadBalancerHostname()
+	if len(hostname) == 0 {
+		// If the retrieved address is empty or nil, clean up and exit.
+		log.Infoln("Cleaning up check and exiting because the load balancer hostname is nil.")
+		errorReport := []string{} // Make a slice for errors here, because there can be more than 1 error.
+		// Clean up the check. A deployment was brought up, but no ingress was created.
+		cleanUpError := cleanUp()
+		if cleanUpError != nil {
+			errorReport = append(errorReport, cleanUpError.Error())
+		}
+		hostnameError := errors.New("service load balancer ingress hostname is nil: " + hostname)
+		// Report errors to Kuberhealthy and exit.
+		errorReport = append(errorReport, hostnameError.Error())
+		reportErrorsToKuberhealthy(errorReport)
+		return nil
+	}
+
+	// Make an HTTP request to the load balancer for the service at the external IP address.
+	// Utilize a backoff loop for the request, the hostname needs to be allotted enough time
+	// for the hostname to resolve and come up.
+	select {
+	case err := <-makeRequestToDeploymentCheckService(hostname):
+		if err != nil {
+			// Handle errors when the HTTP request process completes.
+			log.Errorln("error occurred creating service in cluster:", err)
+			reportErrorsToKuberhealthy([]string{err.Error()})
+			return nil
+		}
+		// Continue with the check if there is no error.
+		log.Infoln("Successfully hit service endpoint.")
+	case <-ctx.Done():
+		// If there is a cancellation interrupt signal, exit.
+		log.Infoln("Cancelling check and shutting down due to interrupt.")
+		return nil
+	case <-runTimeout:
+		// If requests to the hostname endpoint for a status code of 200 took too long, exit.
+		reportErrorsToKuberhealthy([]string{"failed to create deployment within timeout"})
+		return nil
+	}
+
+	// If a rolling update is enabled, perform a rolling update on the service.
+	if rollingUpdate {
+
+		log.Infoln("Rolling update option is enabled.  Performing roll.")
+
+		// Create a rolling-update deployment resource.
+		rolledUpdateConfig := createDeploymentConfig(true)
```

Comment:
change `createDeploymentConfig` to not take a bool and instead do the one-time modification of the deployment inline here

Reply:
This now takes a string